common/pulp_docker/common/tarutils.py

```diff
@@ -114,3 +114,24 @@ def get_youngest_children(metadata):
                 pass
 
     return list(image_ids)
+
+
+def get_image_manifest(tarfile_path):
+    """
+    Given a path to a tarfile, which itself is the product of "skopeo copy"
+    this returns the decoded manifest.json file if it exists.
+
+    :param tarfile_path: full path to tarfile
+    :type tarfile_path:  basestring
+    :return:             decoded manifest json
+    :rtype               list or dict
+    """
+    image_manifest = []
```

Comment:
you just stated here that V1 images do not contain manifest file https://github.com/pulp/pulp_docker/pull/203#discussion_r132869715

The concept of manifest was introduced with V2 api.
These are the specs of V1 content https://github.com/moby/moby/blob/master/image/spec/v1.md
Yes there are json files, metadata which contains info of each layer.
The structure of each layers looks like:
```
├── 5785b62b697b99a5af6cd5d0aabc804d5748abbb6d3d07da5d1d3795f2dcc83e
│   ├── VERSION
│   ├── json
│   └── layer.tar
```
In the tar of V1 content you will not be able to find `manifest.json` file, so making `image_manifest` an array does not really makes sense( to me, if i am not missing anything)
That said, you will never hit this condition while parsing tar of V1:
```
if os.path.basename(member.path) == 'manifest.json':
```
So you can easily set it to None, because you would have 2 outcomes
1.) while parsing tar V2 image, you would a manifest.json and return it
2.) while parsing tar V1 image, you will not find manifest.json and return None.